src/data/int/basic.lean

```diff
@@ -1022,12 +1022,25 @@ by rw [bit_val, nat.bit_val]; cases b; refl
 @[simp] lemma bodd_bit (b n) : bodd (bit b n) = b :=
 by rw bit_val; simp; cases b; cases bodd n; refl
 
+@[simp] lemma bodd_bit0 (n : ℤ) : bodd (bit0 n) = ff := bodd_bit ff n
+
+@[simp] lemma bodd_bit1 (n : ℤ) : bodd (bit1 n) = tt := bodd_bit tt n
+
 @[simp] lemma div2_bit (b n) : div2 (bit b n) = n :=
 begin
   rw [bit_val, div2_val, add_comm, int.add_mul_div_left, (_ : (_/2:ℤ) = 0), zero_add],
   cases b, all_goals {exact dec_trivial}
 end
 
+lemma bit0_ne_bit1 (m n : ℤ) : bit0 m ≠ bit1 n :=
```

Comment:
This is true in an arbitrary char_zero (semi)ring, right?

Reply:
No, `rat` is a char zero ring but `bit0 (1/2) = bit1 0`

Reply:
Ooh, right. I'm thinking too much in terms of actual numerals.